### PR TITLE
Gate avx2 instructions support for parquet-go

### PR DESCRIPTION
**Pull request by javierhonduco**
Some CPUs / hypervisors don't have avx2 support. This PR gates using
this feature in the parquet-go library during compilation depending
on whether the CPU has this feature

Thanks, @achille-roussel for the hint! (https://github.com/segmentio/parquet-go/issues/189#issuecomment-1147093453) 😄 

This is a remediation for: https://github.com/parca-dev/parca/issues/1062

## Test plan 

- With no avx2 support
```shell
$ make 
[...]
go build -tags purego -o bin/ ./cmd/parca
```

Ran the binary for several minutes with parca-agent sending profiles without any issues

- With avx2 support
```shell
$ make 
[...]
go build  -o bin/ ./cmd/parca
```

@jorgelbg could you please confirm if this fixes the issue for you as well? Thanks!

### CODE REVIEW

**Review by kakkoyun — APPROVED** [review]
LGTM :rocket:

**Comment by brancz** [comment]
I don't think we should do this. This means that all the binaries we publish, which is what most people use, will not take advantage of the vectorized instructions even if they have them. Most users do not build from source. I think we should fix this in parquet-go.

**Comment by achille-roussel** [comment]
I agree with @brancz, I'm sorry if my suggestion led to confusion here. When I brought it up I imagined it might be useful to unblock the particular build that was being blocked on the crash.

**Comment by brancz** [comment]
No worries, I imagined it was something like that.

**Comment by javierhonduco** [comment]
Makes sense, closing this PR

**Pull request closed without merging**
